aea/crypto/cosmos.py

```diff
@@ -499,29 +547,24 @@ def get_deploy_transaction(  # pylint: disable=arguments-differ
             },
         }
         tx = self._get_transaction(
-            account_number,
-            chain_id,
-            tx_fee,
-            denom,
-            gas,
-            memo,
-            sequence,
-            msg=deploy_msg,
+            account_number, chain_id, tx_fee, denom, gas, memo, sequence, msg=store_msg,
         )
         return tx
 
-    def get_init_transaction(
+    def _get_init_transaction(
         self,
         deployer_address: Address,
+        denom: str,
+        chain_id: str,
+        account_number: int,
+        sequence: int,
+        amount: int,
         code_id: int,
         init_msg: Any,
-        amount: int,
-        tx_fee: int,
-        gas: int = 80000,
-        denom: Optional[str] = None,
         label: str = "",
+        tx_fee: int = 0,
+        gas: int = 80000,
```

Comment:
Not new in the PR, but just wondering if this default `gas` value should be pulled from config somewhere?

Reply:
It can be set both in the kwargs and on the ledger connection. We could remove the default value here for both this and the tx_fee.